lib/FilesHooks.php

```diff
@@ -878,9 +928,11 @@ protected function unshareFromGroup(IShare $share) {
 
 		$offset = 0;
 		$users = $group->searchUsers('', self::USER_BATCH_SIZE, $offset);
+		$users = array_map(fn (IUser $user) => $user->getUID(), $users);
 		$shouldFlush = $this->startActivityTransaction();
 		while (!empty($users)) {
-			$this->addNotificationsForGroupUsers($users, $actionUser, $share->getNodeId(), $share->getNodeType(), $share->getTarget(), (int)$share->getId());
+			$userIds = array_map(fn (IUser $user) => $user->getUID(), $users);
```

Comment:
$users contains string[] after line 931, then you try to convert to string again.
Results in:
```
Argument #1 ($user) must be of type OCP\IUser, string given in file 'server/server/apps/activity/lib/FilesHooks.php' line 935
```

Reply:
(or remove 934 to keep the activity transaction as short as possible)